lib/rest/RESTClient.js

```diff
@@ -0,0 +1,3506 @@
+            options.message.sticker_ids = options.message.stickerIDs;
+        }
+
+        return this.post(Endpoints.THREAD_WITHOUT_MESSAGE(channelID), {
+            auth: true,
+            body: {
+                auto_archive_duration: options.autoArchiveDuration,
+                applied_tags: options.appliedTags,
+                invitable: options.invitable,
+                message: options.message,
+                name: options.name,
+                type: options.type,
+                rate_limit_per_user: options.rateLimitPerUser
+            },
+            files: files
+        }).then((channel) => Channel.from(channel, this.#client));
+    }
+
+    /**
+    * Create a thread with an existing message
+    * @arg {String} channelID The ID of the channel
+    * @arg {String} messageID The ID of the message to create the thread from
+    * @arg {Object} options The thread options
+    * @arg {Number} [options.autoArchiveDuration] Duration in minutes to automatically archive the thread after recent activity, either 60, 1440, 4320 or 10080
+    * @arg {String} options.name The thread channel name
+    * @arg {Number} [options.rateLimitPerUser] The ratelimit of the channel, in seconds. 0 means no ratelimit is enabled
+    * @returns {Promise<NewsThreadChannel | PublicThreadChannel>}
+    */
+    async createThreadWithMessage(channelID, messageID, options) {
+        if(this.#client === undefined) {
+            throw new Error("Missing required option \"client\". Please provide a valid \"client\" option in the constructor.");
+        }
+
+        return this.post(Endpoints.THREAD_WITH_MESSAGE(channelID, messageID), {
+            auth: true,
+            body: {
+                name: options.name,
+                auto_archive_duration: options.autoArchiveDuration,
+                rate_limit_per_user: options.rateLimitPerUser
+            }
+        }).then((channel) => Channel.from(channel, this.#client));
+    }
+
+    /**
+     * Crosspost (publish) a message to subscribed channels
+     * @arg {String} channelID The ID of the NewsChannel
+     * @arg {String} messageID The ID of the message
+     * @returns {Promise<Message>}
+     */
+    async crosspostMessage(channelID, messageID) {
+        if(this.#client === undefined) {
+            throw new Error("Missing required option \"client\". Please provide a valid \"client\" option in the constructor.");
+        }
+
+        return this.post(Endpoints.CHANNEL_CROSSPOST(channelID, messageID), {
+            auth: true
+        }).then((message) => new Message(message, this.#client));
+    }
+
+    /**
+     * Makes a DELETE request to the API.
+     * @arg path The endpoint to make the request to.
+     * @arg options Data regarding the request.
+     * @returns Resolves with the returned JSON data.
+     */
+    async delete(path, options) {
+        return this.#handler.request("DELETE", path, options);
+    }
```

Comment:
on `RESTClient#[http_verb]` methods: probably better off providing a generalized `request(method, path, options)` calling the underlying request handler (it will  also be easier to replace by subclasses, should that ever be needed)